### DISCUSSION TITLE
Midi Start & Stop messages in Reaper

### DISCUSSION THREAD

**Discussion by LloydPhillips22** [helgoboss/helgobox#1359] [Q&A] [answered]
I posted a thread in the "General" forum but failed to solve this issue & I wondered if it might be a compatibility problem (Realearn / Helgobox issue is included as that was suggested but has its own issue)
===========
I've got a 1995 Yamaha CBX-K1 that transmits <<FA>>, <<FB>>, <<FC>> midi messages (Start, Stop, Continue).
The messages are being sent (verified with OBDS MIDI Monitor).

As I was unable to find a way to map those directly to Reaper, it was suggested to use using Realearn/Helgobox.

I was able to map the Midi clock start & stop messages, and The "Midi Clock Start" message works fine in starting the transport playing, but the midi clock stop message does not stop the transport...but instead, returns to zero and continues playing!
Playing then can not be stopped until I uncheck the [enable/disable] box.

The learn source detects and names the midi clock start & stop properly, but maps the target for BOTH to:
"Project: Invoke transport action Play/Stop" for BOTH.

I'm not sure why this fails and maybe a clue is that Reaper fails to map to the actual "Play" & "Stop" actions, as I chose in this picture?

![Transport](https://github.com/user-attachments/assets/e13bc7cd-fc21-44b5-ac71-b5c9304b0540)

__________________
Computer: Win10, I7 7700, 32Gig - Equipment: Yamaha CBX-K1, Keith McMillan QuNexus

**Comment by helgoboss** [comment, marked as answer]
You used ReaLearn's own target "Project: Invoke transport action" with action "Play/stop" (this doesn't execute a REAPER action, so it doesn't directly map to any of the actions on the right!). The way this target works is that it plays when receiving a value > 0% and stop when receiving a 0% value. The MIDI clock sources always send 100%.

Solution: Set "Target min" and "Target max" to 0% in the second mapping.

Alternative: Use target "Project: Invoke REAPER action" and specifically choose the "Transport: Stop" action.

**Reply by LloydPhillips22** [reply]
Thanks you so much!  
I realize now I was quite confused when I posted, and did not even mention the "Midi Clock Continue" mapping that was giving me the real trouble.
I'd just clicked the "Learn Target" box and then selected the actions from Reaper's Action list.
I had not even gotten to HelgoBox's "EDIT" button at that point.
Those actions with low-numbered IDs (1007, 1008, 1016,etc) seem to invoke a "transport" action in HelgoBox as opposed to a "Reaper action".
I notice the default shortcuts to those in Reaper are "media keyboard".

But those particular midi clock Start & Stop messages actually work fine just using the default detected transport functions or by selecting a Reaper action.
I could not find another "Stop" action in Reaper, and I did try out all the "Play/Stop"(40044) & "Play/Pause" (40073) Actions, and those DO show as Reaper Actions.      

- For "Midi Clock Start" the only result i've been able to get is a 'start / stop' function, whether ReaLearn target or Reaper action "Transport: Play". 
   That is no big deal to me, and actually removes the "Midi Clock Stop" KEY free for other use.  
- "Midi Clock Stop" seems to function as expected, stopping the transport (and returning to the start location).

- "Midi Clock Continue" is one I have trouble with.
If I set it to "Project: Invoke REAPER action" and then choose 1008 "Transport Pause", it causes the transport to become stuck, moving forward only in tiny jerks.
(I notice that it automatically selects "Absolute 14-bit" as the trigger. I tried all the options but none seem to eliminate the 'sticking' transport.

I tried choosing (40328) Transport: Play/Stop (move edit cursor on stop) and that gives the same problem.
So I tried to use the trigger 2. % (%), and set both to 0%, but then nothing happens at all.
Interestingly, while both are set to 0% , allowing the transport to play, I can use the "Trigger!" and it works perfectly, pausing the transport and moving the edit cursor.
Same is true if I disable the mapping, and use the 'Trigger!" button.

The "PAUSE" action in Reaper normally keeps the Transport's "Play" arrow green and toggles the "Pause" button on and off, and the "Trigger" button does that as well, but in any other case, the "Play"&"Pause" both just flash briefly and the transport moves just a tiny bit.

I also see that the "Midi Clock Continue" -> "Reaper Action "Transport Pause" makes the mapping to 'Midi CLock Stop" flash as well, as if it is both unpausing and stopping at the same time.
Perhaps it is something to do with the fact that I do not find an actual "Continue" Reaper Action. 
And the "Continue" message might be tied to the Song Position Pointer or something ?

I have a video I can link...  
https://drive.google.com/file/d/1pK0z3Vm_hXE2Slo0yr_KsFCgL-n_94TJ/view?usp=drive_link

**Reply by helgoboss** [reply]
Let's not focus on the targets. The targets work as they should, pretty sure. Just two rules of thumb: If you use a normal (not on/off-style) REAPER action, you should use invocation type "Trigger". And the Glue section should be the default (press "Reset").

Whatever issue you have, it's related to either your Yamaha device or ReaLearn's MIDI clock source (rarely used, I think) or the way you configured your mapping. Here's how you can investigate your issues:

1. Press Menu => Logging => Log real control messages.
2. Then use your Yamaha device to trigger the MIDI clock messages. What do you see?
3. Ideally, you would see single messages popping up whenever you start/stop/pause/continue playback on your Yamaha device. Nothing repetitive.

Post the result here.
